### PR TITLE
feat(actions): build the binaries and new release

### LINKED ISSUE #672: Automate new release builds with GitHub Actions
cf. https://github.com/Scalingo/link/pull/127 for reference

### PR DESCRIPTION
**Pull request by EtienneM**
~Waiting for #737 to be merged~

It is actually hard to test that the release creation works, without actually triggering a release 😅 

Soooooo let's actually merge this PR and wait for the next tag push

Fix #672

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -28,4 +28,16 @@ jobs:
       - run: go version
 
       - name: Execute the tests
-        run: go test -race -p=1 ./...
```
Comment (EtienneM) [inline .github/workflows/publish.yml:31]: We want to execute the tests in parallel

File: dists/make-release.sh [hunk 2]
```diff
@@ -2,18 +2,15 @@
 
 [ "$DEBUG" = "1" ] && set -x
 set -e
+set -x
 
 VERSION=""
-BUILD_ONLY=false
```
Comment (EtienneM) [inline dists/make-release.sh:7]: The mode with `BUILD_ONLY` equals to `false` has never been used. Moreover I want we move away from this script (https://github.com/Scalingo/cli/issues/739) so it's not really an issue to delete a bit of code here.

File: dists/make-release.sh [hunk 3]
```diff
@@ -25,12 +21,6 @@ fi
 bin_dir="bin/$VERSION"
 mkdir -p $bin_dir
 
-read -p "Which Rollbar token should be used in this release: " ROLLBAR_TOKEN
-if [[ -z $ROLLBAR_TOKEN ]]; then
-  echo "Rollbar token is mandatory" >&2
-  exit 2
-fi
```
Comment (EtienneM) [inline dists/make-release.sh:32]: Rollbar is actually never used:

https://github.com/Scalingo/cli/blob/81aeb22b58168fc08865e8786ebdd64aced3c395/cmd/error.go#L41-L57

And I'm not sure we want to use it at some point...

File: dists/make-release.sh [hunk 4]
```diff
@@ -106,20 +79,3 @@ fi
...
-  echo "- Set the title to v${VERSION}"
-  echo "- Set the branch to ${VERSION}"
-  echo "- Add all zip and tar.gz from bin/${VERSION}"
-  echo "- Set the content to:"
-  sed -n "/$VERSION/,/$current_version/p" CHANGELOG.md | sed -e '1d;$d'
-  echo -e "\n\nOnce done, restart the cli-download-service with"
-  echo "scalingo --region osc-fr1 -a cli-download-service restart"
-fi
-
-exit 1
```
Comment (EtienneM) [inline dists/make-release.sh:125]: Why would you make the script execution fails?.. 😅

**Review by brandon-welsch — APPROVED** [review]
LGTM

**Pull request merged**